src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -286,18 +286,16 @@ await onMessage(messageContext)
 
                 var errorContext = new ErrorContext(exception, headers, messageId, body, transportTransaction, processingFailures);
 
-                // the transport tests assume that all transports use a circuit breaker to be resilient against exceptions
-                // in onError. Since we don't need that robustness, we just retry onError once should it fail.
                 ErrorHandleResult actionToTake;
                 try
                 {
                     actionToTake = await onError(errorContext)
                         .ConfigureAwait(false);
                 }
-                catch (Exception)
+                catch (Exception ex)
                 {
-                    actionToTake = await onError(errorContext)
-                        .ConfigureAwait(false);
+                    criticalError.Raise($"Failed to execute recoverability policy for message with native ID: `{messageContext.MessageId}`", ex);
```

Comment:
I'm not sure what I think about the change to have the word "native" in there. What is that trying to convey?

Reply:
That the ID is not the NServiceBus "Message ID" but rather the ID assigned by the queueing system.

Eg. for MSMQ the ID would be  `72802d3f-fa83-4088-8d02-8489a341c6e2\23577` but the NSB ID is just a GUID.

Does that make sense?

Reply:
What about for transports that don't assign a message id like that?